File: src/main/java/de/pflugradts/passbird/application/commandhandling/command/CommandType.java
package de.pflugradts.passbird.application.commandhandling.command;

import de.pflugradts.passbird.domain.model.transfer.Bytes;
import lombok.Getter;

import java.util.Arrays;

public enum CommandType {

    CUSTOM_SET('c'),
    DISCARD('d'),
    EXPORT('e'),
    GET('g'),
    HELP('h'),
    IMPORT('i'),
    LIST('l'),
    NAMESPACE('n'),
    QUIT('q'),
    RENAME('r'),
    SET('s'),
    VIEW('v'),
    UNDEFINED('?');

    @Getter
    private final char type;

    CommandType(final char type) {
        this.type = type;
    }

    public static CommandType fromCommandBytes(final Bytes type) {
        return type.isEmpty() ? UNDEFINED : Arrays.stream(CommandType.values())
            .filter(commandType -> commandType
                    .name()
                    .toLowerCase()
                    .charAt(0) == type.getFirstByte())
            .findAny()
            .orElse(UNDEFINED);
    }

}

File: src/main/kotlin/de/pflugradts/passbird/application/commandhandling/CommandType.kt
package de.pflugradts.passbird.application.commandhandling

import de.pflugradts.passbird.domain.model.shell.Shell

enum class CommandType(val type: Char) {
    CUSTOM_SET('c'),
    DISCARD('d'),
    EXPORT('e'),
    FAVORITE('f'),
    GET('g'),
    HELP('h'),
    IMPORT('i'),
    KEYSTORE('k'),
    LIST('l'),
    MEMORY('m'),
    NEST('n'),
    PROTEIN('p'),
    QUIT('q'),
    REPEAT('.'),
    RENAME('r'),
    SET('s'),
    VIEW('v'),
    YOLK('y'),
    UNDEFINED('?'),
    ;

    companion object {
        fun resolveCommandTypeFrom(commandShell: Shell) = if (commandShell.isNotEmpty) {
            entries.find { it.type.code.toByte() == commandShell.firstByte } ?: UNDEFINED
        } else {
            UNDEFINED
        }
    }
}
